standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/properties/Digester.java

```diff
@@ -0,0 +1,440 @@
+     * @param str the string
+     * @return this digester
+     */
+    public Digester digest(String str) {
+        if (str != null) {
+            final int sz = str.length();
+            for(int i = 0; i < sz; ++i) {
+                digest(str.charAt(i));
+            }
+        }
+        return this;
+    }
+
+    /**
+     * Updates the digest with a uuid.
+     * @param uid the uuid
+     * @return this digester
+     */
+    public Digester digest(UUID uid) {
+        digest(md, uid);
+        return this;
+    }
+
+    /**
+     * Updates the digest with a uuid.
+     * @param uri the uri
+     * @return this digester
+     */
+    public Digester digest(URI uri) {
+        digest(md, uri.toString());
+        return this;
+    }
+
+    /**
+     * Updates the digest with an object that describes how it digests.
+     * @param digestible the object
+     * @return this digester
+     */
+    public Digester digest(Digestible digestible) {
+        if (digestible != null) {
+            digestible.digest(this);
+        }
+        return this;
+    }
+
+    /**
+     * Updates the digest with a stream.
+     * @param input the stream
+     * @return this digester
+     */
+    public Digester digest(InputStream input) {
+        if (input != null) {
+            digest(md, input);
+        }
+        return this;
+    }
+    /**
+     * Updates the digest with any (serializable) object.
+     * @param obj the object
+     * @return this digester
+     */
+    public Digester digest(Object obj) {
+        if (obj instanceof Digestible) {
+            return digest((Digestible) obj);
+        }
+        if (obj instanceof UUID) {
+            return digest((UUID) obj);
+        }
+        if (obj instanceof URI) {
+            return digest((URI) obj);
+        }
+        if (obj instanceof String) {
+            return digest((String) obj);
+        }
+        if (obj instanceof Date) {
+            return digest((Date) obj);
+        }
+        if (obj instanceof Integer) {
+            return digest(((Integer) obj).intValue());
+        }
+        if (obj instanceof Long) {
+            return digest(((Long) obj).longValue());
+        }
+        if (obj instanceof Double) {
+            return digest(((Double) obj).doubleValue());
+        }
+        if (obj instanceof Boolean) {
+            return digest(((Boolean) obj).booleanValue());
+        }
+        if (obj instanceof Character) {
+            return digest(((Character) obj).charValue());
+        }
+        if (obj instanceof Short) {
+            return digest(((Short) obj).intValue());
+        }
+        if (obj instanceof Float) {
+            return digest(((Float) obj).doubleValue());
+        }
+        if (obj != null && obj.getClass().isArray()) {
+           int sz  = java.lang.reflect.Array.getLength(obj);
+           for(int i = 0; i < sz; ++i) {
+               digest(java.lang.reflect.Array.get(obj, i));
+           }
```

Comment:
nit: maybe we need to add `return this` after processing the `array`

Reply:
Good catch! corrected.